java/client/src/main/java/glide/connectors/handlers/ChannelHandler.java

```diff
@@ -22,9 +22,10 @@ public class ChannelHandler {
 
     protected final Channel channel;
     protected final CallbackDispatcher callbackDispatcher;
+    private boolean isClosed = false;
```

Comment:
Use `AtomicBoolean` which is thread safe

Reply:
Note: except that the variable is bi-directional.  It doesn't matter if multiple threads set `isClosed` to `true`, it will be `true`.